src/main/java/controllers/ProjectInputController.java

```diff
@@ -0,0 +1,309 @@
+                projectDeleteMember(projectToManage, projectCommand);
+            } else if (projectCommand.length() == 12 && ("view members").equals(projectCommand)) {
+                projectViewMembers(projectToManage);
+            } else if (projectCommand.length() == 12 && ("view credits").equals(projectCommand)) {
+                projectViewCredits();
+            } else if (projectCommand.length() >= 9 && ("add task ").equals(projectCommand.substring(0, 9))) {
+                projectAddTask(projectToManage, projectCommand);
+            } else if (projectCommand.length() >= 10 && ("view tasks").equals(projectCommand.substring(0,10))) {
+                projectViewTasks(projectToManage, projectCommand);
+            } else if (projectCommand.length() == 19 && ("view assigned tasks").equals(projectCommand)) {
+                projectViewAssignedTasks(projectToManage.getAssignedTaskList());
+            } else if (projectCommand.length() > 25
+                    && ("view task requirements i/").equals(projectCommand.substring(0, 25))) {
+                projectViewTaskRequirements(projectToManage, projectCommand);
+            } else if (projectCommand.length() > 23
+                    && ("edit task requirements ").equals(projectCommand.substring(0, 23))) {
+                projectEditTaskRequirements(projectToManage, projectCommand);
+            } else if (projectCommand.length() >= 10 && ("edit task ").equals(projectCommand.substring(0, 10))) {
+                projectEditTask(projectToManage, projectCommand);
+            } else if (projectCommand.length() >= 12 && ("delete task ").equals(projectCommand.substring(0,12))) {
+                projectDeleteTask(projectToManage, projectCommand);
+            } else if (projectCommand.length() >= 12 && ("assign task ").equals(projectCommand.substring(0,12))) {
+                projectAssignTask(projectToManage, projectCommand);
+            } else if ("bye".equals(projectCommand)) {
+                consoleView.end();
+            } else {
+                consoleView.consolePrint("Invalid command. Try again!");
+            }
+        } else {
+            consoleView.consolePrint("Please enter a command.");
+        }
+        return isManagingAProject;
+    }
+
+    /**
+     * Adds a member to the current project.
+     * @param projectToManage The project specified by the user.
+     * @param projectCommand The user input.
+     */
+    public void projectAddMember(Project projectToManage, String projectCommand) {
+        String memberDetails = projectCommand.substring(11);
+        int numberOfCurrentMembers = projectToManage.getNumOfMembers();
+        memberDetails = memberDetails + " x/" + numberOfCurrentMembers;
+        IMember newMember = memberFactory.create(memberDetails);
+        if (newMember.getName() != null) {
+            projectToManage.addMember((Member) newMember);
+            consoleView.addMember(projectToManage, newMember.getDetails());
```

Comment:
The lower level tiers must not be aware of higher level tiers (remember n-tier architecture?). The view should 'ask' for data and get a 'response' in return.  

Try to enforce that pattern.